### ISSUE TITLE
Support apt-get (auto)clean

### ISSUE THREAD

**Issue by tersmitten** [ansible/ansible-modules-core#2340]
https://help.ubuntu.com/community/AptGet/Howto

> `apt-get autoclean`: This command removes .deb files for packages that are no longer installed on your system. Depending on your installation habits, removing these files from /var/cache/apt/archives may regain a significant amount of diskspace
> 
> `apt-get clean`: The same as above, except it removes all packages from the package cache. This may not be desirable if you have a slow Internet connection, since it will cause you to redownload any packages you need to install a program.

Maybe `du -sb /var/cache/apt/archives` can be used to check idempotency?

See related #1976

**Comment by tersmitten** [comment]
I'm working on more states: https://github.com/tersmitten/ansible-modules-core/compare/support-apt-get-autoclean

**Comment by jimi-c** [comment]
As this is already requested as a feature, I'm going to go ahead and close this.

If you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by rfdparker** [comment]
Was this closed because of #1976?

That issue appears to deal with `apt-get autoremove` but not `apt-get clean`.

It's important to understand that `apt-get clean`, `apt-get autoclean` and `apt-get autoremove` are all quite different from one another.

To quote the `apt-get(8)` man page:
```
clean
    clean clears out the local repository of retrieved package files.
    It removes everything but the lock file from
    /var/cache/apt/archives/ and /var/cache/apt/archives/partial/.

autoclean (and the auto-clean alias since 1.1)
    Like clean, autoclean clears out the local repository of retrieved
    package files. The difference is that it only removes package files
    that can no longer be downloaded, and are largely useless. This
    allows a cache to be maintained over a long period without it
    growing out of control. The configuration option
    APT::Clean-Installed will prevent installed packages from being
    erased if it is set to off.

autoremove (and the auto-remove alias since 1.1)
    autoremove is used to remove packages that were automatically
    installed to satisfy dependencies for other packages and are now no
    longer needed.
```

As you can see, while `autoremove` relates to removing installed packages, `clean` clears the local repository of package *files*.

If Ansible's `apt` module supported `apt-get clean` it would be very useful for cleaning-up that local repository of package files which can otherwise grow in size over time.  As the man page suggests I don't think `apt-get autoremove` is particularly useful however.
